### ISSUE TITLE
Support new CNI capabilities.

### ISSUE THREAD

**Issue by Random-Liu** [containerd/cri#1243]
There are several useful new capabilities we should support https://github.com/containernetworking/cni/blob/d19c2358a56d66cff6694e4f00f7c38c16d8182e/CONVENTIONS.md#well-known-capabilities:
* ipRanges: We should pass the pod cidr down there;
* dns: This is especially important for windows support.

**Comment by Random-Liu** [comment]
`ipRanges` requires the cni plugin to provide pod cidr when creating pods.

However, kubelet only pass down the pod cidr once during node startup, after containerd restarts we'll lose the pod cidr.

The only solution is to checkpoint the cidr somewhere, which I prefer not to do unless necessary. 

Dockershim doesn't have this problem, because it has the same lifecycle with kubelet. And after kubelet restarts it will send pod cidr to the container runtime again. https://github.com/kubernetes/kubernetes/commit/5d9ec20d7e2c4beb7c0d7eb3533cd2a179fff479

In conclusion, I prefer not supporting `ipRanges` which requires an extra checkpoint until we have to.
